### PR TITLE
Use the '__assign' helper for transforms

### LINKED ISSUE #8180: [Transforms] __assign polyfill needed to replace React.__spread
From the typescript: 

```ts
//// [react-consumer.tsx]
namespace M {
	export var React: any;
}

namespace M {
	// Should emit M.React.createElement
	//  and M.React.__spread
	var foo;
	var spread1 = <div x='' {...foo} y='' />;

	// Quotes
	var x = <div>This "quote" thing</div>;
}
```

Expected an __assign polyfill:

```js
//// [react-consumer.js]
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var M;
(function (M) {
})(M || (M = {}));
var M;
(function (M) {
    // Should emit M.React.createElement
    //  and M.React.__spread
    var foo;
    var spread1 = M.React.createElement("div", __assign({ x: '' }, foo, { y: '' }));
    // Quotes
    var x = M.React.createElement("div", null, "This \"quote\" thing");
})(M || (M = {}));
```

Actual, a call to React.__spread instead:

```js
//// [file.js]
//// [react-consumer.js]
var M;
(function (M) {
})(M || (M = {}));
var M;
(function (M) {
    // Should emit M.React.createElement
    //  and M.React.__spread
    var foo;
    var spread1 = M.React.createElement("div", M.React.__spread({ x: '' }, foo, { y: '' }));
    // Quotes
    var x = M.React.createElement("div", null, "This \"quote\" thing");
})(M || (M = {}));
```

Failed tests:
1. tsxReactEmit 2, 4, 5 and 6
2. tsxExternalModule2
3. reactNamespaceJSXEmit

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #8180.

### CODE REVIEW

File: src/compiler/factory.ts [hunk 1]
```diff
@@ -822,16 +822,6 @@ namespace ts {
         return react;
     }
 
-    export function createReactSpread(reactNamespace: string, segments: Expression[], parentElement: JsxOpeningLikeElement) {
-        return createCall(
-            createPropertyAccess(
-                createReactNamespace(reactNamespace, parentElement),
```
Comment (sandersn) [inline src/compiler/factory.ts]: is `createReactNamespace` still used now?

Comment (sandersn) [inline src/compiler/factory.ts]: Yes, in `createReactCreateElement`

**Comment by sandersn** [comment]
:+1:

File: src/compiler/factory.ts (hunk 1 above)
Comment (DanielRosenwasser) [inline src/compiler/factory.ts]: Yes, for `React.createElement`

**Pull request merged**